include/ginkgo/core/base/types.hpp

```diff
@@ -488,6 +488,57 @@ GKO_ATTRIBUTES constexpr bool operator!=(precision_reduction x,
     template _macro(std::complex<double>, int64)
 
 
+#define GKO_INSTANTIATE_FOR_EACH_MIXED_VALUE_AND_INDEX_TYPE(_macro) \
```

Comment:
I would prefer to have this macro in the `ell_kernels` header (and named something with `INSTANTIATE_ELL_FOR`...) since it is currently an ELL exclusive.

Reply:
I would like to keep it here as this will hopefully generalize to more matrix formats in the future

Reply:
Can you add a small name change to it, so it is clear that you need to have 3 different type template parameter? The current name is not really clear how many parameters you need for the macro.

Reply:
The other reason why I would like to have this in the `ell_kernels` header is because this is a public header, meaning we have to keep it in here unchanged until we have Ginkgo 2.0.

Reply:
Looking at this huge list, I would definitely agree with @upsj that we should allow disabling of these instantiations and also disable them by default.